File: src/main/java/pl/bk/news/api/dto/NewsDto.java
package pl.bk.news.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class NewsDto
{
    String country;
    String category;
    List<ArticleDto> articles;
}

File: client/src/app/dto/news-dto.ts
import {ArticleDto} from "./article-dto";

export class NewsDto {

  country: String;
  category: String;
  articles: ArticleDto[] = [];

  constructor(data: any) {
    this.country = data.country;
    this.category = data.category;
    Array.from(data.articles).forEach(entry => {
      this.articles.push(new ArticleDto(entry))
    });
  }
}
